armi/reactor/converters/tests/test_axialExpansionChanger_MultiPin.py

```diff
@@ -67,295 +69,293 @@ def setUp(self):
+            toComp.getMass(),
+            toCompRefData.mass + self.amountBeingRedistributed,
             places=self.places,
         )
-        # assert that the temperature of c0 is the same and that c1 has increased
-        self.assertEqual(self.c0.temperatureInC, preRedistributionC0Temp)
-        self.assertGreater(self.c1.temperatureInC, preRedistributionC1Temp)
 
-        # now remove the c0 mass and assert it is deltaZTop/self.c0.height less than its pre-redistribution value
-        self.axialExpChngr.removeMassFromComponent(fromComp=self.c0, deltaZTop=deltaZTop)
+        # assert that the temperature of fromComp is the same
+        self.assertEqual(fromComp.temperatureInC, fromCompRefData.temp)
+        # assert toComp temp based on if thermal expansion
+        if thermalExp:
+            self.assertGreater(toComp.temperatureInC, toCompRefData.temp)
+        else:
+            self.assertEqual(toComp.temperatureInC, toCompRefData.temp)
+
+    def _rmMassFromCompWithTempAssert(self, fromComp: Component):
+        """Remove the mass from ``fromComp``.
+
+        Notes
+        -----
+        Two assertions are done: 1) the correct amount of mass is moved to ``toComp``. 2) the resulting temperatures
+        for ``fromComp`` and ``toComp`` are correct.
+        """
+        # remove mass from ``fromComp``
+        self.axialExpChngr._removeMassFromComponent(fromComp=fromComp, deltaZTop=self.deltaZTop)
+
+        fromCompRefData, _toCompRefData = self._getReferenceData(fromComp, None)
+        # ensure the fromComp mass decreases by amountBeingRedistributed
+        self._updateFromCompElevations(fromComp)
         self.assertAlmostEqual(
-            self.c0.getMass(), preRedistributionC0Mass - amountBeingRedistributed, places=self.places
+            fromComp.getMass(), fromCompRefData.mass - self.amountBeingRedistributed, places=self.places
         )
-        # assert the c0 temperature does not change
-        self.assertEqual(self.c0.temperatureInC, preRedistributionC0Temp)
+        # assert the fromComp temperature does not change
+        self.assertEqual(fromComp.temperatureInC, fromCompRefData.temp)
 
 
 class TestMultiPinConservation(TestMultiPinConservationBase):
     def setUp(self):
         super().setUp()
-        _origBHeight, _origCMassesByBlock, self.origTotalCMassByFlag = self.getMassesForTest(self.a)
+        self.origTotalCMassByFlag = self.getTotalCompMassByFlag(self.a)
 
     @staticmethod
-    def getMassesForTest(a: "HexAssembly"):
-        @dataclass
-        class StoreMass:
-            cFlags: str
-            mass: float
-
-        blockHeights: dict["HexBlock", float] = {}
-        compMassByBlock: dict["HexBlock", StoreMass] = collections.defaultdict(list)
+    def _isFluidButNotBond(c):
+        """Determine if a component is a fluid, but not Bond."""
+        return bool(isinstance(c, Component) and isinstance(c.material, Fluid) and not c.hasFlags(Flags.BOND))
```

Comment:
These should already be boolean and don't need the re-bool cast
```suggestion
        return isinstance(c, Component) and isinstance(c.material, Fluid) and not c.hasFlags(Flags.BOND)
```